include/deal.II/sundials/arkode.h

```diff
@@ -0,0 +1,858 @@
+   * proportionately to $m_k N$, larger values of $m_k$ may result in faster
+   * convergence.
+   *
+   * This improvement may be significant even for "small" values, e.g. $1 \leq
+   * m_k \leq 5$, and convergence may not improve (or even deteriorate) for
+   * larger values of $m_k$. While ARKode uses a Newton-based iteration as its
+   * default solver due to its increased robustness on very stiff problems, it
+   * is highly recommended that users also consider the fixed-point solver for
+   * their cases when attempting a new problem.
+   *
+   * For either the Newton or fixed-point solvers, it is well-known that both
+   * the efficiency and robustness of the algorithm intimately depends on the
+   * choice of a good initial guess. In ARKode, the initial guess for either
+   * nonlinear solution method is a predicted value $z_i(0)$ that is computed
+   * explicitly from the previously-computed data (e.g. $y_{n−2}, y_{n−1}$, and
+   * $z_j$ where $j < i$). Additional information on the specific predictor
+   * algorithms implemented in ARKode is provided in ARKode documentation.
+   *
+   * The user has to provide the implementation of the following std::function:
+   *  - reinit_vector;
+   * and either one or both of
+   *  - implicit_function;
+   *  - explicit_function;
+   *
+   * If the mass matrix is different from the identity, the user should supply
+   *  - solve_mass_system;
+   * and, optionally,
+   *  - setup_mass;
+   *
+   * If the use of a Newton method is desired, then the user should also supply
+   *  - solve_jacobian_system;
+   * and optionally
+   *  - setup_jacobian;
+   *
+   * Also the following functions could be rewritten. By default
+   * they do nothing, or are not required.
+   *  - solver_should_restart;
+   *  - get_local_tolerances;
+   *
+   * To produce output at fixed steps, overload the function
+   *  - output_step;
+   *
+   *
+   * To provide a simple example, consider the harmonic oscillator problem:
+   * \f[
+   * \begin{split}
+   *   u'' & = -k^2 u \\
+   *   u (0) & = 0 \\
+   *   u'(0) & = k
+   * \end{split}
+   * \f]
+   *
+   * We write it in terms of a first order ode:
+   *\f[
+   * \begin{matrix}
+   *   y_0' & =  y_1 \\
+   *   y_1' & = - k^2 y_0
+   * \end{matrix}
+   * \f]
+   *
+   * That is $y' = A y$
+   * where
+   * \f[
+   * A:=
+   * \begin{matrix}
+   * 0 & 1 \\
+   * -k^2 &0
+   * \end{matrix}
+   * \f]
+   * and $y(0)=(0, k)$.
+   *
+   * The exact solution is $y_0(t) = \sin(k t)$, $y_1(t) = y_0'(t) = k \cos(k t)$,
+   * $y_1'(t) = -k^2 \sin(k t)$.
+   *
+   * A minimal implementation, using only explicit RK methods, is given by the
+   * following code snippet:
+   *
+   * @code
+   * typedef Vector<double> VectorType;
+   *
+   * SUNDIALS::ARKode<VectorType> ode;
+   *
+   * ode.reinit_vector = [&] (VectorType&v)
+   * {
+   *   v.reinit(2);
+   * };
+   *
+   * double kappa = 1.0;
+   *
+   * ode.explicit_function = [&] (double,
```

Comment:
it would make me more comfortable if you did `[kappa]` because people are otherwise going to refactor this into something where you keep a reference to a local variable